cesium/internal/unary/iterator.go

```diff
@@ -65,7 +65,7 @@ func (i *Iterator) SeekLast(ctx context.Context) bool {
 		return false
 	}
 	ok := i.internal.SeekLast(ctx)
-	i.seekReset(i.internal.TimeRange().End)
+	i.seekReset(i.internal.TimeRange().BoundBy(i.bounds).End)
```

Comment:
No and agreed – the only other places that use `seekReset` are `SeekLE` and `SeekGE` – which would return false on a call to an out-of-bounds timestamp. We should not allow the view to go out of bounds ever. I will move the check for bounds into the `seekReset` method

Reply:
NVM – since one of them needs the Start of a timerange and the other one the End, I don't think it makes sense to add that part to the logic of `seekReset`, I will change the SeekGE and SeekLE calls to disallow out-of-bound timestamps though

Reply:
Sorry what cleanup? Code cleanup or cleaning up an actual resource in the iterator? a bit confused